src/test/java/uk/gov/companieshouse/web/pps/validation/EnterDetailsValidatorTest.java

```diff
@@ -0,0 +1,104 @@
+    void isValidWhenRefStartsWithLateFilingAndMissing() {
+        enterDetails.setPenaltyReferenceName(LATE_FILING.name());
+        enterDetails.setPenaltyRef("");
+        BindingResult bindingResult = new BeanPropertyBindingResult(enterDetails, ENTER_DETAILS_MODEL);
+
+        testValidator.isValid(enterDetails, bindingResult);
+
+        assertTrue(bindingResult.hasErrors());
+        assertEquals(1, bindingResult.getFieldErrorCount(PENALTY_REF_FIELD));
+        assertEquals("Enter the reference number", Objects.requireNonNull(bindingResult.getFieldError(PENALTY_REF_FIELD)).getDefaultMessage());
+    }
+
+    @Test
+    void isValidWhenRefStartsWithLateFilingAndNotValid() {
+        enterDetails.setPenaltyReferenceName(LATE_FILING.name());
+        enterDetails.setPenaltyRef("A123456");
+        BindingResult bindingResult = new BeanPropertyBindingResult(enterDetails, ENTER_DETAILS_MODEL);
+
+        testValidator.isValid(enterDetails, bindingResult);
+
+        assertTrue(bindingResult.hasErrors());
+        assertEquals(1, bindingResult.getFieldErrorCount(PENALTY_REF_FIELD));
+        assertEquals("Enter your reference number exactly as shown on your penalty letter", Objects.requireNonNull(bindingResult.getFieldError(PENALTY_REF_FIELD)).getDefaultMessage());
+    }
+
+    @Test
+    void isValidWhenRefStartsWithSanctions() {
+        enterDetails.setPenaltyReferenceName(SANCTIONS.name());
+        enterDetails.setPenaltyRef("PN12345678");
+        BindingResult bindingResult = new BeanPropertyBindingResult(enterDetails, ENTER_DETAILS_MODEL);
+
+        testValidator.isValid(enterDetails, bindingResult);
+
+        assertFalse(bindingResult.hasErrors());
+    }
+
+    @Test
+    void isValidWhenRefStartsWithSanctionsAndMissing() {
+        enterDetails.setPenaltyReferenceName(SANCTIONS.name());
+        enterDetails.setPenaltyRef("");
+        BindingResult bindingResult = new BeanPropertyBindingResult(enterDetails, ENTER_DETAILS_MODEL);
+
+        testValidator.isValid(enterDetails, bindingResult);
+
+        assertTrue(bindingResult.hasErrors());
+        assertEquals(1, bindingResult.getFieldErrorCount(PENALTY_REF_FIELD));
+        assertEquals("Enter the penalty reference", Objects.requireNonNull(bindingResult.getFieldError(PENALTY_REF_FIELD)).getDefaultMessage());
+    }
+
+    @Test
+    void isValidWhenRefStartsWithSanctionsAndNotValid() {
+        enterDetails.setPenaltyReferenceName(SANCTIONS.name());
+        enterDetails.setPenaltyRef("PN1234567");
+        BindingResult bindingResult = new BeanPropertyBindingResult(enterDetails, ENTER_DETAILS_MODEL);
+
+        testValidator.isValid(enterDetails, bindingResult);
+
+        assertTrue(bindingResult.hasErrors());
+        assertEquals(1, bindingResult.getFieldErrorCount(PENALTY_REF_FIELD));
+        assertEquals("Enter your penalty reference exactly as shown on your penalty letter", Objects.requireNonNull(bindingResult.getFieldError(PENALTY_REF_FIELD)).getDefaultMessage());
```

Comment:
Could you split this onto separate line please?

Reply:
Done in commit [41d34dc](https://github.com/companieshouse/penalty-payment-web/pull/180/commits/41d34dc611b21f63dfeb40c27c5693f97fbf140a)